smithy-model/src/test/java/software/amazon/smithy/model/transform/DowngradeToV1Test.java

```diff
@@ -120,15 +120,34 @@ public void removesResourceProperties() {
     public void removesUnnecessaryDefaults() {
         String stringModel = "$version: \"2.0\"\n"
                 + "namespace smithy.example\n"
+                // Root level shapes with a 1.0 non-zero value drop the default value and are boxed
                 + "@default(10)\n"
                 + "integer MyInteger\n"
+                // Root level shapes with a 1.0 zero value shape keeps the default value
+                + "@default(0)\n"
+                + "integer ZeroInteger\n"
+                // Root level shapes with no default value are boxed
+                + "integer BoxedInteger\n"
+                // StructureShape still exists
                 + "structure Struct {\n"
+                // A Member that targets a shape with a 1.0 non-zero value drops the default value and are not boxed
                 + "  @default(10)\n"
                 + "  foo: MyInteger\n"
+                // A Member that targets a shape with a 1.0 zero value and the member default value matches keeps the default value
+                + "  zeroTargetZeroMember: ZeroInteger = 0\n"
+                // A Member that has a default value of null keeps the default value and is boxed
+                + "  zeroTargetBoxedMember: ZeroInteger = null\n"
+                // A Member that has doesn't have the same default value on the member and trait have the default value dropped
```

Comment:
This sentence doesn't make sense. It seems like
```
// A Member that has a zero default value and targets a shape without a default value...
```

Which also tells me that we should add a case for 
```
// A Member that has a non-zero default value and targets a shape without a default value...
boxedTargetZeroMember: BoxedInteger = 20\n"
```